.github/workflows/benchmark.yaml

```diff
@@ -75,59 +75,61 @@ jobs:
       - run-benchmark
 
     steps:
-    - name: "Install Rust toolchain"
-      run: rustup show
-
-    - name: "Install critcmp"
-      # Use debug build: Building takes much longer than the "slowness" of using the debug build.
-      run: cargo install --debug critcmp
-
-    - name: "Linux | Download PR benchmark results"
-      uses: actions/download-artifact@v3
-      with:
-        name: benchmark-results-ubuntu-latest
-        path: ./target/criterion
-
-    - name: "Linux | Compare benchmark results"
-      shell: bash
-      run: |
-        echo "### Benchmark" >> summary.md
-        echo "#### Linux" >> summary.md
-        echo "\`\`\`" >> summary.md
-        critcmp main pr >> summary.md
-        echo "\`\`\`" >> summary.md
-        echo "" >> summary.md
-
-    - name: "Linux | Cleanup benchmark results"
-      run: rm -rf ./target/criterion
-
-    - name: "Windows | Download PR benchmark results"
-      uses: actions/download-artifact@v3
-      with:
-        name: benchmark-results-windows-latest
-        path: ./target/criterion
-
-    - name: "Windows | Compare benchmark results"
-      shell: bash
-      run: |
-        echo "#### Windows" >> summary.md
-        echo "\`\`\`" >> summary.md
-        critcmp main pr >> summary.md
-        echo "\`\`\`" >> summary.md
-        echo "" >> summary.md
-
-        echo ${{ github.event.pull_request.number }} > pr-number
-
-        cat summary.md > $GITHUB_STEP_SUMMARY
-
-    - uses: actions/upload-artifact@v3
-      name: Upload PR Number
-      with:
-        name: pr-number
-        path: pr-number
-
-    - uses: actions/upload-artifact@v3
-      name: Upload Summary
-      with:
-        name: summary
-        path: summary.md
+      - name: "Install Rust toolchain"
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@v2
```

Comment:
I considered adding rust-cache to the critcmp step but decided against it because GitHub limits the cache per repository to 10GB. That means GitHub starts evicting cached artifacts once we exceed the 10GB limit. You can see all our cache artifacts [here](https://github.com/charliermarsh/ruff/actions/caches). 

Now, the reason I excluded `critcmp` was to avoid adding yet another cache (where each cache is around 600MB) that results in the eviction of other cached artifacts (e.g., for the clippy process) because I thought having warm caches for clippy is more important than for the benchmark results. 

I don't know if that's the right call, nor have I verified my assumption that caching the c`ritcmp` build causes any build time regression for other jobs. 

I did some more research to find ways to avoid building `critcmp` altogether. I was excited when I read that [criterion 0.4](https://www.tweag.io/blog/2022-03-03-criterion-rs/) has built-in tabulating. No need to use `critcmp`, we can generate the comparison directly in the benchmark step. Unfortunately, they [removed tabulating](https://github.com/bheisler/criterion.rs/pull/610) before the release in favor of adding it to [`cargo-criterion`](https://github.com/bheisler/cargo-criterion). I tried `cargo criterion` locally, but the feature doesn't seem to exist, and even if it does, we would only be switching one dependency with another. 

So what if we could avoid running `cargo install` altogether? Can we download the `critcmp` binary or cache it somehow? That's when I discovered [`cargo-binstall`](https://github.com/cargo-bins/cargo-binstall). It installs binaries rather than building the crates from source... yay! Uhm, okay, but how do we install `cargo-binstall`?

Luckily `cargo-binstall` publishes prebuilt binaries on GitHub and there are even a github actions to download binaries from the release page ([release-downloader](https://github.com/marketplace/actions/release-downloader) or [`dtonlay/install`](https://github.com/dtolnay/install)). Alternatively, we could just use `curl` or `wget` to get the binary. Do you want to give this at try?

Reply:
Yes!

Reply:
This shouldn't be necessary as far as I understand. You can use `cargo binstall` to install `critcmp`. It uses the [`quickinstall`](https://github.com/cargo-bins/cargo-quickinstall) registry under the hood that provides pre-built binaries (at least it works on my linux machine)